### PR TITLE
Bonnel : LED Sensor & Effecter PDR support in PLDM

### PR DESCRIPTION
**Pull request by vkaverap**
Signed-off-by: Varsha Kaverappa <vkaverap@in.ibm.com>

### CODE REVIEW

File: oem/ibm/configurations/pdr/ibm,bonnel/11.json [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+                "path": "/xyz/openbmc_project/led/groups/fan1_identify",
+                "interface": "xyz.openbmc_project.Led.Group",
+                "property_name": "Asserted",
+                "property_type": "bool",
+                "property_values" : [false, true]
+             }
+        }]
+    },
+    {
+        "entity_path" : "/xyz/openbmc_project/inventory/system/chassis/motherboard/nvme0",
```
Comment (jaypadath) [inline oem/ibm/configurations/pdr/ibm,bonnel/11.json:59]: This path should be "/xyz/openbmc_project/inventory/system/chassis/motherboard/dasd_backplane/nvme0". Please correct for all nvme paths.

**Review by jaypadath — CHANGES_REQUESTED** [review]
Other comments:
1. Folder name should be "ibm,bonnell" instead of "ibm,bonnel"
2. Op-panel frus are missing. I see those in Jinu's commit
    "/xyz/openbmc_project/inventory/system/chassis/motherboard/panel0"
     "/xyz/openbmc_project/inventory/system/chassis/motherboard/panel1"
     You could add both and later remove if not needed while testing.

File: oem/ibm/configurations/pdr/ibm,bonnel/11.json (hunk 1 above)
Comment (vkaverap) [reply oem/ibm/configurations/pdr/ibm,bonnel/11.json]: done

File: oem/ibm/configurations/pdr/ibm,bonnell/4.json [hunk 2]
```diff
@@ -0,0 +1,420 @@
...
+        "instance" : 1,
+        "container" : 1,
+        "sensors" : [{
+            "set" : {
+                "id" : 17,
+                "size" : 1,
+                "states" : [1,2]
+            },
+            "dbus" : {
+                "path": "/xyz/openbmc_project/led/groups/enclosure_identify",
```
Comment (jaypadath) [inline oem/ibm/configurations/pdr/ibm,bonnell/4.json:16]: All the dbus details are wrong. This should match with "/xyz/openbmc_project/led/physical/virtual_enc_id". Please check the 4.json file at "pldm/oem/ibm/configurations/pdr/ibm,rainier-1s4u" as an example.

File: oem/ibm/configurations/pdr/ibm,bonnell/11.json [hunk 3]
```diff
@@ -0,0 +1,420 @@
...
+    },
+    {
+        "entity_path" : "/xyz/openbmc_project/inventory/system/chassis/motherboard/fan0",
+        "effecters" : [{
+            "set" : {
+                "id" : 10,
+                "size" : 1,
+                "states" : [1,2]
+            },
+            "dbus" : {
```
Comment (jaypadath) [inline oem/ibm/configurations/pdr/ibm,bonnell/11.json:238]: The dbus details are wrong in all the below paths in 11.json file.  Please check the 11.json file at "pldm/oem/ibm/configurations/pdr/ibm,rainier-1s4u" as an example.

So for fan0 it should be like below:
            "dbus" : {
                "path": "/xyz/openbmc_project/inventory/system/chassis/motherboard/fan0",
                "interface": "xyz.openbmc_project.State.Decorator.OperationalStatus",
                "property_name": "Functional",
                "property_type": "bool",
                "property_values" : [true, false]
             }

File: oem/ibm/configurations/pdr/ibm,bonnell/4.json [hunk 4]
```diff
@@ -0,0 +1,420 @@
...
+        "type" : 45,
+        "instance" : 1,
+        "container" : 1,
+        "sensors" : [{
+            "set" : {
+                "id" : 10,
+                "size" : 1,
+                "states" : [1,2]
+            },
+            "dbus" : {
```
Comment (jaypadath) [inline oem/ibm/configurations/pdr/ibm,bonnell/4.json:221]: All the dbus details are wrong. This should match with "/xyz/openbmc_project/led/physical/virtual_enc_fault". Please check the 4.json file at "pldm/oem/ibm/configurations/pdr/ibm,rainier-1s4u" as an example.

**Review by jaypadath — CHANGES_REQUESTED** [review]

File: oem/ibm/configurations/pdr/ibm,bonnell/11.json [hunk 5]
```diff
@@ -0,0 +1,420 @@
...
+    {
+        "entity_path" : "/xyz/openbmc_project/inventory/system/chassis/motherboard/dasd_backplane/nvme0",
+        "effecters" : [{
+            "set" : {
+                "id" : 10,
+                "size" : 1,
+                "states" : [1,2]
+            },
+            "dbus" : {
+                "path": "/xyz/openbmc_project/inventory/system/chassis/motherboard/nvme0",
```
Comment (jaypadath) [inline oem/ibm/configurations/pdr/ibm,bonnell/11.json:307]: this should match with entity_path

File: oem/ibm/configurations/pdr/ibm,bonnell/11.json [hunk 6]
```diff
@@ -0,0 +1,420 @@
...
+    {
+        "entity_path" : "/xyz/openbmc_project/inventory/system/chassis/motherboard/dasd_backplane/nvme1",
+        "effecters" : [{
+            "set" : {
+                "id" : 10,
+                "size" : 1,
+                "states" : [1,2]
+            },
+            "dbus" : {
+                "path": "/xyz/openbmc_project/inventory/system/chassis/motherboard/nvme1",
```
Comment (jaypadath) [inline oem/ibm/configurations/pdr/ibm,bonnell/11.json:324]: this should match with entity_path

File: oem/ibm/configurations/pdr/ibm,bonnell/11.json [hunk 7]
```diff
@@ -0,0 +1,420 @@
...
+    {
+        "entity_path" : "/xyz/openbmc_project/inventory/system/chassis/motherboard/dasd_backplane/nvme2",
+        "effecters" : [{
+            "set" : {
+                "id" : 10,
+                "size" : 1,
+                "states" : [1,2]
+            },
+            "dbus" : {
+                "path": "/xyz/openbmc_project/inventory/system/chassis/motherboard/nvme2",
```
Comment (jaypadath) [inline oem/ibm/configurations/pdr/ibm,bonnell/11.json:341]: this should match with entity_path

File: oem/ibm/configurations/pdr/ibm,bonnell/11.json [hunk 8]
```diff
@@ -0,0 +1,420 @@
...
+    {
+        "entity_path" : "/xyz/openbmc_project/inventory/system/chassis/motherboard/dasd_backplane/nvme3",
+        "effecters" : [{
+            "set" : {
+                "id" : 10,
+                "size" : 1,
+                "states" : [1,2]
+            },
+            "dbus" : {
+                "path": "/xyz/openbmc_project/inventory/system/chassis/motherboard/nvme3",
```
Comment (jaypadath) [inline oem/ibm/configurations/pdr/ibm,bonnell/11.json:358]: this should match with entity_path

File: oem/ibm/configurations/pdr/ibm,bonnell/4.json [hunk 9]
```diff
@@ -0,0 +1,423 @@
...
+        "type" : 45,
+        "instance" : 1,
+        "container" : 1,
+        "sensors" : [{
+            "set" : {
+                "id" : 17,
+                "size" : 1,
+                "states" : [1,2]
+            },
+            "dbus" : {
```
Comment (jaypadath) [inline oem/ibm/configurations/pdr/ibm,bonnell/4.json:15]: this should be like below
            "dbus" : {
                "path": "/xyz/openbmc_project/led/physical/virtual_enc_id",
                "interface": "xyz.openbmc_project.Led.Physical",
                "property_name": "State",
                "property_type": "string",
                "property_values": [
                         "xyz.openbmc_project.Led.Physical.Action.Off",
                         "xyz.openbmc_project.Led.Physical.Action.On || xyz.openbmc_project.Led.Physical.Action.Blink"
                ]
             }

File: oem/ibm/configurations/pdr/ibm,bonnell/4.json [hunk 10]
```diff
@@ -0,0 +1,423 @@
...
+        "type" : 45,
+        "instance" : 1,
+        "container" : 1,
+        "sensors" : [{
+            "set" : {
+                "id" : 10,
+                "size" : 1,
+                "states" : [1,2]
+            },
+            "dbus" : {
```
Comment (jaypadath) [inline oem/ibm/configurations/pdr/ibm,bonnell/4.json:224]: This should be like below:
            "dbus" : {
                "path": "/xyz/openbmc_project/led/physical/virtual_enc_fault",
                "interface": "xyz.openbmc_project.Led.Physical",
                "property_name": "State",
                "property_type": "string",
                "property_values": [
                         "xyz.openbmc_project.Led.Physical.Action.Off",
                         "xyz.openbmc_project.Led.Physical.Action.On || xyz.openbmc_project.Led.Physical.Action.Blink"
                ]
             }

**Review by jaypadath — CHANGES_REQUESTED** [review]

**Review by jaypadath — COMMENTED** [review]
Changes looks good. Please add the commit message with a reference to Jira story.

**Pull request closed without merging**